### DISCUSSION TITLE
Step 1: Installing dependencies -- my project has no Package.swift file

### DISCUSSION THREAD

**Discussion by andrew-shoonya** [apple/pkl#366] [Q&A] [answered]
**TL;DR**: The very first line of the Quick Start leaves me stuck. **Question**: Is it the case that the pkl-swift library is something that can 

1. only be added as a dependency to an existing Swift package, and
2. not added as a dependency to an existing app? 

Bear in mind that I know nothing about Pkl, but I have been writing compilers on Macs for 30 years, so go ahead and talk tech. 

**Details**: The Quick Start, at [https://pkl-lang.org/swift/current/quickstart.html] starts quickly with:

> 1. Install dependencies. Add the pkl-swift library as a dependency to _your project’s `Package.swift`_:

Emphasis added to stress this: my project doesn't have a `Package.swift` file. In fact, _no project_ has a `Package.swift` file. Nothing you create in Xcode with **New >  Project...** will have a `Package.swift` file. (See end for list of projects I created.) 

Now if I use **New > Package...** (whose existence implies that a package is not a project), then of course there is a `Package.swift` file present. This leads to my **question**: Am I to assume that the pkl-swift library is something that can only be added as a dependency to an existing package, not directly to an existing app project? 

I am sure I would understand the answer to this question if I knew more about the architecture, intent, and usage of Pkl. But I don't, that's why I'm at the Quick Start page. 

* These Projects have no Package.swift file when created: iOS apps, iOS games, iOS static libraries, macOS apps, macOS command lines, macOS and iOS Frameworks and static libraries. I looked all over the web for instructions on how to _add_ a `Package.swift` file to an _existing_ project, but all I get is unneeded advice on how to add a Package _per se_ to an existing project.

**Comment by ktoso** [comment, marked as answer]
Packages are the standard way to distributed swift code really.

- You can depend on packages in Xcode projects like this: https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app - though you should be aware that iOS [isn't ](https://github.com/apple/pkl-swift/commit/28d1782e50f8867ba05925de1c18e86cd09260d6) a supported platform as of right now.
- Or in a package project you'd add it to Package.swift, if you're using a package project

I don't think the package here needs to go in depth about explaining xcode projects because it is primarily aimed at server ecosystem, in which developers generally use the SwiftPM project structure, and do have a Package.swift.

**Reply by andrew-shoonya** [reply]
I have been using Xcode for 20 years. I specifically said (last sentence of my post) that I didn't need advice on how to add a Package to an app. You gave me that advice anyway. 

I specifically asked what things are the target of Step 1 "Add the pkl-swift library as a dependency". You didn't answer that question. 

To reiterate my question: Step 1 of the Quick Start, namely

> Install dependencies. Add the pkl-swift library as a dependency to your project’s Package.swift:

is impossible  to follow, because macOS apps _do not have_ a Package.swift file, because _they are not packages_. It looks like I can only "Add the pkl-swift library as a dependency" to an existing Package, not to an existing app. Is this correct? It seems limiting.

**Reply by StefMa** [reply]
Instead of being passive aggressive you should better be a little bit nice to people who tries to help you. Maybe with that approach you get people who are silent observers to that discussion be actively participating....

**Reply by ktoso** [reply]
> It looks like I can only "Add the pkl-swift library as a dependency" to an existing Package, not to an existing app. Is this correct? It seems limiting.

This is not correct and I have replied as to how to add the dependency into an xcodeproject based build (first link). 

This wording is pretty standard across all swift packages in the ecosystem; I don't think we need to change anything here, especially as this package is _server focused_ where indeed packages are they way to go by default.
